### PR TITLE
Handle registration activation email sending via notify scripts

### PR DESCRIPTION
**Pull request by finmaster19**
Disable the default user registration activation email sending.
Extend the RegistrationReceived event so that it can include
activation url. Then this event can be used to send customized
activation email to registered user.

### CODE REVIEW

File: shuup/front/apps/registration/notify_events.py [hunk 1]
```diff
@@ -13,14 +14,15 @@
...
-from shuup.notify.typology import Email, Model
+from shuup.notify.typology import Email, Model, URL
 
 
 class RegistrationReceived(Event):
     identifier = "registration_received"
     name = _("Registration Received")
     customer = Variable(_("Customer"), type=Model("shuup.Contact"))
     customer_email = Variable(_("Customer Email"), type=Email)
+    activation_url = Variable(_("Activation URL"), type=URL, required=False)
```
Comment (suutari-ai) [inline shuup/front/apps/registration/notify_events.py]: Missing translations. 1.2 should be fully translated to at least English, Finnish and Swedish, so please update the corresponding PO files too.

**Comment by suutari-ai** [comment]
Hmm.. there was a part that disabled e-mail sending of django-registration-redux, but I don't see it anymore. Shouldn't the e-mails of django-registration-redux be disabled?

**Comment by suutari-ai** [comment]
This should have a test that checks that the event is really sent and it contains a correct activation URL.

**Comment by finmaster19** [comment]
@suutari-ai Yes this is missing translations and tests. I opened a pr so I can get feedback about this feature. I was thinking that if we still leave the possibility to send activation email. Now setting `SEND_ACTIVATION_EMAIL` to False disables automatic activation email sending and then you could make your own notify script which handles it or not.

**Comment by suutari-ai** [comment]
But if SEND_ACTIVATION_EMAIL is not set to False and merchant defines a notification script that sends an e-mail from the event, doesn't it then send two e-mails (one from django-registration-redux and one from Shuup's notifications)?

Though it's a bit problematic to just set it to False, because then shops upgrading Shuup from pre-1.2 to 1.2 would lose their activation e-mails. Any ideas how to address that?

One option would be to have a default notification script installed by a data migration, but I'm not sure if that's good, since installing default data is currently handled by shuup_init, not data migrations.

**Comment by finmaster19** [comment]
> But if SEND_ACTIVATION_EMAIL is not set to False and merchant defines a notification script that sends an e-mail from the event, doesn't it then send two e-mails (one from django-registration-redux and one from Shuup's notifications)?

With this approach it would send two different email if SEND_ACTIVATION_EMAIL is True and you create your own notify script which also sends the link. When SEND_ACTIVATION_EMAIL is True, registration-redux sends automatic activation email. Then you don't need to create your own notify script which sends activation email, though you can use this `RegistrationReceived` how you want to create other scripts. But if SEND_ACTIVATION_EMAIL is False, then no automatic activation is send. Then you should at least create one script which uses this event and sends activation link.

**Comment by suutari-ai** [comment]
Yeah, but if the code sends the hard coded activation email anyway, then you can't really customize what's being sent to the customer.

File: shuup/front/apps/registration/locale/fi_FI/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -45,6 +45,9 @@ msgstr "Asiakas"
 msgid "Customer Email"
 msgstr "Asiakkaan sähköpostiosoite"
 
+msgid "Activation URL"
+msgstr "Aktivointi URL"
```
Comment (suutari-ai) [inline shuup/front/apps/registration/locale/fi_FI/LC_MESSAGES/django.po]: Aktivointi-URL

**Comment by finmaster19** [comment]
Updated this pr so that you can use `user_is_active` event field as condition when script is run and activation email is sent.

**Comment by suutari-ai** [comment]
Great work, Aleksi! Thanks!

**Pull request merged**